esp/esp/program/modules/handlers/admincore.py

```diff
@@ -83,6 +84,69 @@ def main(self, request, tl, one, two, module, extra, prog):
 
         return render_to_response(self.baseDir()+'directory.html', request, context)
 
+    @aux_call
+    @needs_admin
+    def settings(self, request, tl, one, two, module, extra, prog):
+        from esp.program.modules.forms.admincore import ProgramSettingsForm, TeacherRegSettingsForm, StudentRegSettingsForm
+        context = {}
+        submitted_form = ""
+        crmi = ClassRegModuleInfo.objects.get(program=prog)
+        scrmi = StudentClassRegModuleInfo.objects.get(program=prog)
+        old_url = prog.url
+
+        #If one of the forms was submitted, process it and save if valid
+        if request.method == 'POST':
+            if 'form_name' in request.POST:
+                submitted_form = request.POST['form_name']
+                if submitted_form == "Program Settings":
+                    form = ProgramSettingsForm(request.POST, instance = prog)
+                    prog_form = form
+                elif submitted_form == "Teacher Registration Settings":
+                    form = TeacherRegSettingsForm(request.POST, instance = crmi)
+                    crmi_form = form
+                elif submitted_form == "Student Registration Settings":
+                    form = StudentRegSettingsForm(request.POST, instance = scrmi)
+                    scrmi_form = form
+                if form.is_valid():
+                    form.save()
+                    #If the url for the program is now different, redirect to the new settings page
+                    if prog.url is not old_url:
+                        return HttpResponseRedirect( '/manage/%s/settings' % (prog.url))
+
+        #Set up any other forms on the page
+        if submitted_form is not "Program Settings":
```

Comment:
Why do you use `is not` here (and in several lines below) rather than `==`?  You want string equality, not pointer equality.

Reply:
Whoops, must have gotten my python and django template logic mixed up here. Thanks for catching it!